snakemake/path_modifier.py

```diff
@@ -52,9 +52,18 @@ def modify(self, path, property=None):
                 modified_path = AnnotatedString(modified_path)
             modified_path.flags.update(path.flags)
             if is_flagged(modified_path, "multiext"):
-                modified_path.flags["multiext"] = self.apply_default_storage(
-                    self.replace_prefix(modified_path.flags["multiext"], property)
-                )
+                # if multiext is just str, update it as such.
+                if isinstance(modified_path.flags["multiext"], str):
+                    modified_path.flags["multiext"] = self.apply_default_storage(
+                        self.replace_prefix(modified_path.flags["multiext"], property)
+                    )
```

Comment:
I don't think the if case is needed. In the multiext function above you always (correctly) set the flag value to the MultiextValue object.

Reply:
It's still needed, at least for the CI tests (test_module_complex and test_output_file_cache_storage with having the modified_path coming as MultiExtValue (i.e. before set_input/set_output) and str (after set_input/set_output), respectively). Not sure if this is the most appropriate place to fix this, though..